File: solutions/ocaml/raindrops/1/raindrops.ml
let raindrop_factors = [(3, "Pling");(5, "Plang");(7, "Plong");];;

let raindrop n =
  let result =
    raindrop_factors
    |> List.map (fun (k, v) -> if (n mod k = 0) then v else "")
    |> String.concat ""
  in
  if result = "" then string_of_int n else result 

File: solutions/elixir/raindrops/2/lib/raindrops.ex
defmodule Raindrops do
  @raindrops %{
    3 => "Pling",
    5 => "Plang",
    7 => "Plong"
  }
  @doc """
  Returns a string based on raindrop factors.

  - If the number contains 3 as a prime factor, output 'Pling'.
  - If the number contains 5 as a prime factor, output 'Plang'.
  - If the number contains 7 as a prime factor, output 'Plong'.
  - If the number does not contain 3, 5, or 7 as a prime factor,
    just pass the number's digits straight through.
  """
  @spec convert(pos_integer) :: String.t()
  def convert(number) do
    @raindrops
    |> Enum.map(fn {k, v} -> if Integer.mod(number, k) === 0, do: v, else: "" end)
    |> List.to_string()
    |> then(&if String.length(&1) > 0, do: &1, else: to_string(number))
  end
end
